Source/UI/NewUI/BillingForm/View/Field/BillingFormPhoneNumberText.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+import PhoneNumberKit
+
+protocol BillingFormPhoneNumberTextDelegate {
+    func updateCountryCode(code:Int)
+}
+
+final class BillingFormPhoneNumberText: BillingFormTextField {
+    var phoneNumberTextDelegate: BillingFormPhoneNumberTextDelegate?
+    var phoneNumber: PhoneNumber?
+    var nationalNumber: String {
+        let rawNumber = self.text ?? ""
+        return partialFormatter.nationalNumber(from: rawNumber)
+    }
+    var isValidNumber: Bool {
+        let rawNumber = self.text ?? ""
+        do {
+            phoneNumber = try phoneNumberKit.parse(rawNumber.replacingOccurrences(of: " ", with: ""))
+            return true
+        } catch {
+            return false
+        }
+    }
+    
+    private var previousTextCount = 0
+    private var previousFormat = ""
+    private let phoneNumberKit = PhoneNumberKit()
+    private lazy var _defaultRegion: String = PhoneNumberKit.defaultRegionCode()
+    private lazy var partialFormatter = PartialFormatter(phoneNumberKit: phoneNumberKit, defaultRegion: "GB", withPrefix: true)
+
+    init(type: BillingFormCell, tag: Int, phoneNumberTextDelegate: BillingFormPhoneNumberTextDelegate) {
+        super.init(type: type,tag: tag)
+        self.phoneNumberTextDelegate = phoneNumberTextDelegate
+        setup()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setup() {
+        keyboardType = .phonePad
+        updateTextFieldFormat(textField: self)
+        addTarget(self, action: #selector(textFieldDidChange), for: UIControl.Event.editingChanged)
+    }
+
+    /// Called when the text changed.
+    @objc private func textFieldDidChange(textField: UITextField) {
+        updateTextFieldFormat(textField: textField)
+    }
+    
+    private func updateTextFieldFormat(textField: UITextField) {
```

Comment:
Can we add some comments for future devs what this updateTextFieldFormat is actually doing and all the code snippet below ?

Reply:
I have added a comment.
TODO: Copied from old code. Needs to be refactored